wallet/wallet-controller/src/lib.rs

```diff
@@ -175,11 +187,82 @@ impl<T: NodeInterface + Clone + Send + Sync + 'static> Controller<T> {
             .map_err(ControllerError::NodeCallError)
     }
 
+    pub fn create_wallet(
+        &mut self,
+        file_path: impl AsRef<Path>,
+        mnemonic: mnemonic::Mnemonic,
+        passphrase: Option<&str>,
+    ) -> Result<(), ControllerError<T>> {
+        utils::ensure!(
+            !file_path.as_ref().exists(),
+            ControllerError::WalletFileError(
+                file_path.as_ref().to_owned(),
+                "File already exists".to_owned()
+            )
+        );
+
+        let db = wallet::wallet::open_or_create_wallet_file(file_path)
+            .map_err(ControllerError::WalletError)?;
+        let wallet = wallet::Wallet::new_wallet(
+            Arc::clone(&self.chain_config),
+            db,
+            &mnemonic.to_string(),
+            passphrase,
+        )
+        .map_err(ControllerError::WalletError)?;
+
+        self.add_wallet(wallet);
+        Ok(())
+    }
+
+    pub fn open_wallet(&mut self, file_path: impl AsRef<Path>) -> Result<(), ControllerError<T>> {
+        utils::ensure!(
+            file_path.as_ref().exists(),
+            ControllerError::WalletFileError(
+                file_path.as_ref().to_owned(),
+                "File does not exist".to_owned()
+            )
+        );
+
+        let db = wallet::wallet::open_or_create_wallet_file(file_path)
```

Comment:
Same here... open shouldn't create a wallet.

Reply:
We check that the file exists.
Should I move the check to the wallet crate and split `open_or_create_wallet_file` to two functions?

Reply:
For CLI it makes sense to open or create. I'm worried we're gonna have a problem in the GUI where someone tries to create a wallet, then they have an issue where we overwrite their wallet or something.

So, this is OK for the CLI, but for the GUI there has to be a strong distinction, since people usually select wallets manually through a file dialog.

Reply:
I agree. But UIs should use the `wallet-controller` crate to manage wallets, and there are two separate functions specifically for that: `create_wallet` and `open_wallet`. `create_wallet` checks that the file does not exist and `open_wallet` checks that the wallet file exists. So I don't see a problem here.

Reply:
Alrighty. Sounds good.